.github/workflows/release.yml

```diff
@@ -20,34 +31,69 @@ jobs:
           /p:PackageProjectUrl="${{ github.server_url }}/${{ github.repository }}/tree/${{ github.event.release.tag_name }}"
           /p:PackageReleaseNotes="${{ github.server_url }}/${{ github.repository }}/blob/${{ github.event.release.tag_name }}/CHANGELOG.md"
+          ${{ env.version_suffix_args }}
 
-      - name: Test
+      - name: Unit Test
+        run: dotnet test
+          --configuration Release
+          --filter="TestCategory=Smoke&TestCategory!=Manual"
+          --logger "trx;LogFileName=${{ github.workspace }}/artifacts/test-results/smoke.trx"
+          ${{ env.version_suffix_args }}
+          
+      - name: Run Live Tests
         run: dotnet test
           --configuration Release
-          --filter="TestCategory~Online"
-          --logger "trx;LogFileName=${{github.workspace}}/artifacts/test-results/full.trx"
+          --filter="TestCategory!=Smoke&TestCategory!=Images&TestCategory!=Moderations&TestCategory!=Manual"
+          --logger "trx;LogFilePrefix=live"
+          --results-directory ${{ github.workspace }}/artifacts/test-results
+          ${{ env.version_suffix_args }}
         env:
           OPENAI_API_KEY: ${{ secrets.OPENAI_API_KEY }}
 
+      - name: Upload artifact
+        uses: actions/upload-artifact@v2
+        if: ${{ !cancelled() }}
+        with:
+          name: build-artifacts
+          path: ${{ github.workspace }}/artifacts
+
+  deploy:
+    name: Publish Package
+    needs: build
+    if: github.event_name != 'pull_request'
+    runs-on: ubuntu-latest
+    permissions:
+      packages: write
+      contents: write
+    env:
+      NUGET_API_KEY: ${{ github.event_name == 'release' && secrets.NUGET_API_KEY || secrets.GITHUB_TOKEN }}
+      PUBLISH_TARGET: ${{ github.event_name == 'release' && 'https://api.nuget.org/v3/index.json' || 'github' }}
+    steps:
+      - uses: actions/download-artifact@v2
+
       # Append the nuget package to the github release that triggered this workflow
       - name: Upload release asset
+        if: github.event_name == 'release'
         run: gh release upload ${{ github.event.release.tag_name }}
-          ${{github.workspace}}/artifacts/packages/*.*nupkg
+          ${{ github.workspace }}/build-artifacts/packages/*.*nupkg
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
 
-      - name: Upload artifact
-        uses: actions/upload-artifact@v2
-        with:
-          name: build-artifacts
-          path: ${{github.workspace}}/artifacts
+      - name: NuGet Autenticate
+        if: github.event_name != 'release'
+        run: dotnet nuget add source
+          "https://nuget.pkg.github.com/${{ github.repository_owner }}/index.json"
+          --name "github"
+          --username ${{ github.actor }}
+          --password ${{ secrets.GITHUB_TOKEN }}
+          --store-password-in-clear-text
 
-      - name: Publish
+      - name: Publish NuGet package
         run: dotnet nuget push
```

Comment:
This needs a condition of ==release

Reply:
We publish the scheduled runs to the local feed and release to nuget.org.   The conditionals in the job variables pick the feed and token to use